nameko/messaging.py

```diff
@@ -166,26 +182,15 @@ def retry_policy(self):
         return DEFAULT_RETRY_POLICY
 
     def setup(self):
-
-        exchange = self.exchange
-        queue = self.queue
+        if self.exchange is not None:
```

Comment:
Why not doing this in the `__init__` method, since we are already updating `declare` with `queue`?